include/nuttx/net/ioctl.h

```diff
@@ -112,8 +112,12 @@
 
 /* SocketCAN ****************************************************************/
 
-#define SIOCGCANBITRATE  _SIOC(0x002C)  /* Get bitrate from a CAN controller */
-#define SIOCSCANBITRATE  _SIOC(0x002D)  /* Set bitrate of a CAN controller */
+#define SIOCGCANBITRATE    _SIOC(0x002C)  /* Get bitrate from a CAN controller */
+#define SIOCSCANBITRATE    _SIOC(0x002D)  /* Set bitrate of a CAN controller */
+#define SIOCACANEXTFILTER  _SIOC(0x002E)  /* Add hardware-level exteneded ID filter */
```

Comment:
Probably `CAN_RAW_FILTER` has to be mapped to this somehow.

Reply:
According to Linux docs SocketCAN filters are SW only.

Reply:
Indeed Linux SocketCAN is SW only using the CAN_RAW_FILTER socketoption (which is already implemented in NuttX).

It seems that @JacobCrabill added a direct ioctl to the SocketCAN netdevice to set the hardware filters to avoid unnecessary interrupts and can still nicely co-exist with the SocketCAN software filter to filter further towards different userspace applications.

Reply:
SocketCAN has the next:
```
CAN Controller Hardware Filters
To reduce the interrupt load on deep embedded systems some CAN controllers support the filtering of CAN IDs or ranges of CAN IDs. These hardware filter capabilities vary from controller to controller and have to be identified as not feasible in a multi-user networking approach. The use of the very controller specific hardware filters could make sense in a very dedicated use-case, as a filter on driver level would affect all users in the multi-user system. The high efficient filter sets inside the PF_CAN core allow to set different multiple filters for each socket separately. Therefore the use of hardware filters goes to the category ‘handmade tuning on deep embedded systems’. The author is running a MPC603e @133MHz with four SJA1000 CAN controllers from 2002 under heavy bus load without any problems …
```

So @JacobCrabill added an option that is "handmade tuning on deep embedded systems" from SocketCAN point of view